mobile/did_key.go

```diff
@@ -0,0 +1,140 @@
+package mobile
+
+import (
+	"github.com/TBD54566975/ssi-sdk/crypto"
+	"github.com/TBD54566975/ssi-sdk/crypto/jwx"
+	"github.com/TBD54566975/ssi-sdk/did/key"
+	"github.com/goccy/go-json"
+	"github.com/sirupsen/logrus"
+)
+
+// GetSupportedKeyTypes returns a list of supported key types as string values
+func GetSupportedKeyTypes() []string {
+	keyTypes := make([]string, 0, len(key.GetSupportedDIDKeyTypes()))
+	for _, kt := range key.GetSupportedDIDKeyTypes() {
+		keyTypes = append(keyTypes, string(kt))
+	}
+	return keyTypes
+}
+
+type GenerateDIDKeyResult struct {
+	DID string         `json:"did"`
+	JWK map[string]any `json:"jwk"`
+}
+
+// GenerateDIDKey generates a new DID key and returns a JSON representation of GenerateDIDKeyResult
+func GenerateDIDKey(kt string) ([]byte, error) {
+	privateKey, didKey, err := key.GenerateDIDKey(crypto.KeyType(kt))
+	if err != nil {
+		logrus.WithError(err).Error("failed to generate did key")
+		return nil, err
+	}
+
+	expanded, err := didKey.Expand()
+	if err != nil {
+		logrus.WithError(err).Error("failed to expand did key")
+		return nil, err
+	}
+
+	_, jwkPrivateKey, err := jwx.PrivateKeyToPrivateKeyJWK(expanded.VerificationMethod[0].ID, privateKey)
+	if err != nil {
+		logrus.WithError(err).Error("failed to convert private key to jwk")
+		return nil, err
+	}
+
+	jwkBytes, err := json.Marshal(jwkPrivateKey)
+	if err != nil {
+		logrus.WithError(err).Error("failed to marshal jwk")
+		return nil, err
+	}
+
+	var jwk map[string]any
+	if err = json.Unmarshal(jwkBytes, &jwk); err != nil {
+		logrus.WithError(err).Error("failed to unmarshal jwk")
+		return nil, err
+	}
+
+	result := GenerateDIDKeyResult{
+		DID: didKey.String(),
+		JWK: jwk,
+	}
+	return json.Marshal(result)
+}
+
+type CreateDIDKeyRequest struct {
+	KeyType      string         `json:"keyType"`
+	PublicKeyJWK map[string]any `json:"publicKeyJwk"`
+}
+
+type CreateDIDKeyResult struct {
+	DID string `json:"did"`
+}
+
+// CreateDIDKey creates a new DID key from an existing public key, accepting a JSON representation of CreateDIDKeyRequest
+// and returns a JSON representation of CreateDIDKeyResult which contains the DID of the newly created key as a string
+func CreateDIDKey(requestBytes []byte) ([]byte, error) {
```

Comment:
Let's make it easy for clients to understand what they should put into `requestBytes`: can we type this?

Reply:
It's tough to say, because I don't know how a mobile dev would interact with this. I'm of the opinion that a worthy goal is to have mobile devs not have to read any go code at all.

So I ask myself, as a mobile dev, do I know what I should put in here? I think the current answer is probably not. I would make a suggestion if I understood better how calling from Java/Kotlin, Swift would look like.

Reply:
I think go docs are the best we can realistically do...if they don't read the API docs they'll always be lost

Reply:
How about having the type of the parameter this function accepts be `CreateDIDKeyRequest`? According to the [documentation](https://pkg.go.dev/golang.org/x/mobile/cmd/gobind#hdr-Type_restrictions), structs are allowed.

It feels like it's a bit easier to deal with a typed object in Kotlin / Swift.

I was hoping that documentation gets automatically translated into the artifact that's created (the AAR or the Obj-C lib), but it unfortunately doesn't look like that's the case. That's unfortunate :( 
To mitigate, I would say adding an example in the target language that demonstrates how to call this function would be great.

Reply:
will take this in a follow up